src/mongo/db/auth/oidc/oidc_test_fixture.h

```diff
@@ -0,0 +1,203 @@
+    }
+};
+
+// Setter for the 'supportsHumanFlows' field
+struct SetSupportsHumanFlows : public SetFieldWithValue<bool> {
+    using SetFieldWithValue::SetFieldWithValue;
+
+    void set(OidcIdentityProviderConfig& config) override {
+        config.setSupportsHumanFlows(getValue());
+    }
+};
+
+// Setter for the 'JWKSPollSecs' field
+struct SetJWKSPollSecs : public SetFieldWithValue<std::int32_t> {
+    using SetFieldWithValue::SetFieldWithValue;
+
+    void set(OidcIdentityProviderConfig& config) override {
+        config.setJWKSPollSecs(getValue());
+    }
+};
+
+// Creates an OidcIdentityProviderConfig object with the given required parameters
+// and a list of optional field setters.
+template <typename... Setters>
+OidcIdentityProviderConfig create_config(const std::string& issuer,
+                                         const std::string& authNamePrefix,
+                                         const std::string& audience,
+                                         Setters&&... setters) {
+    static_assert((std::conjunction_v<std::is_base_of<SetField, std::decay_t<Setters>>...>),
+                  "All setters must inherit from SetField");
+
+    OidcIdentityProviderConfig config;
+    config.setIssuer(issuer);
+    config.setAuthNamePrefix(authNamePrefix);
+    config.setAudience(audience);
+    (setters.set(config), ...);
+    return config;
+}
+
+// Mock for PeriodicRunner to control the job execution
+// and to verify the job's state.
+class PeriodicRunnerMock : public PeriodicRunner {
+    class ControllableJobMock : public ControllableJob {
+    public:
+        explicit ControllableJobMock(PeriodicJob job) : _job(std::move(job)) {}
+        void start() override {
+            _started = true;
+        }
+
+        void pause() override {
+            FAIL("pause() should not be called");
+        }
+
+        void resume() override {
+            FAIL("resume() should not be called");
+        }
+
+        void stop() override {}
+
+        Milliseconds getPeriod() const override {
+            return _job.interval;
+        }
+
+        void setPeriod(Milliseconds ms) override {
+            _job.interval = ms;
+        }
+
+        bool isStarted() const {
+            return _started;
+        }
+
+    private:
+        bool _started{false};
+        PeriodicJob _job;
+    };
+
+public:
+    PeriodicJobAnchor makeJob(PeriodicJob job) override {
+        auto handle = std::make_shared<ControllableJobMock>(std::move(job));
+        jobs.push_back(handle);
+        return PeriodicJobAnchor{std::move(handle)};
+    }
+
+    std::vector<std::shared_ptr<ControllableJobMock>> jobs;
+};
+
+// Mock for JWKSFetcherFactory to control the creation of JWKSFetcher instances
+struct JWKSFetcherFactoryMock : public JWKSFetcherFactory {
+    class JWKSFetcherMock : public crypto::JWKSFetcher {
+    public:
+        crypto::JWKSet fetch() override {
+            FAIL("JWKSFetcherMock::fetch() is not implemented");
```

Comment:
This method can be implemented when adding unit tests for the `sasl_oidc_server_mechanism` - which I would suggest.